src/App.js

```diff
@@ -1,25 +1,76 @@
-import logo from './logo.svg';
+import React from 'react';
+import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
+// import request from 'superagent';
 import './App.css';
+import Home from './Home/home.js';
+import Login from './Login/login.js';
+import Signup from './Login/signup.js';
+import Todos from './Todo/todo.js';
+import Header from './Components/header.js';
+import {
+  getUserFromLocalStorage,
+  putUserInLocalStorage
+} from './local-storage-utils.js';
+import PrivateRoute from './Components/private-route.js';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
-}
+export default class App extends React.Component {
+  state = {
+    user: getUserFromLocalStorage()
+  };
+
+  handleUserChange = (user) => {
+    this.setState({ user });
+
+    putUserInLocalStorage(user);
+  };
 
-export default App;
+  handleUserLogout = () => {
+    this.handleUserChange();
+  };
+
+  render() {
+    return (
+      <div>
+        <Router>
+          <Header handleUserLogout={this.handleUserLogout} />
```

Comment:
Here we need to pass the `user` to the `Header` component as a prop so the `Header` component can use it in the NavLink logic :)